test/wallet-http-test.js

```diff
@@ -85,6 +92,34 @@ for (const witnessOpt of witnessOptions) {
       addr = Address.fromString(str, node.network);
     });
 
+    it('should change passphrase', async () => {
+      await wallet.setPassphrase('initial');
+
+      // Incorrect Passphrase should not work
+      await assert.rejects(async () => {
+        await wallet.unlock('badpass');
+      }, {
+        name: 'Error',
+        message: 'Could not decrypt.'
+      });
+
+      // Correct Passphrase should work
+      await wallet.unlock('initial');
```

Comment:
Did it work though? Is there any way we can test that the wallet was locked before this line and is now unlocked?

Reply:
line 96: setting the passphrase locked the wallet
line 99: incorrect passphrase didnt work, so wallet is still locked.
line 107: If it doesnt unlock, we will get an error.

We can test whether a wallet is locked or not at any given instant by running a harmless function such as `retoken` which will require a passphrase and throw an error if wallet is locked.